### PR TITLE
(Closed draft) Sites dashboard: Add badge and style tabs

### PR DESCRIPTION
**Pull request by sejas**
#### Proposed Changes

* Fix coming soon filtering
* Style Tabs
* Add new `Badge` component
* [ ] Add the components to Documentation
* [ ] Fix title types, from `string` to `Element`
* [ ] Improve performance adding `useMemo` or equivalent.

#### Screenshots
Screenshot 2022-07-04 at 19 45 05

https://user-images.githubusercontent.com/779993/177200898-77e63925-5bd9-4417-be52-29494e6068de.mp4

#### Testing Instructions

* Access to : /sites-dashboard (https://container-friendly-gauss.calypso.live/sites-dashboard)
* Observe the tabs match the style and functionality.

- Solves #65165

### CODE REVIEW

**Comment by sejas** [comment]
I need to update the typescript types or add comments to ignore the TS validation.
https://github.dev/WordPress/gutenberg/blob/trunk/packages/components/src/tab-panel/index.js#L21-L33

The `TabButton` uses a children to display the content inside itself.

But the types doesn't support a React Element, the title is defined only as string.
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/wordpress__components/tab-panel/index.d.ts#L12

**Comment by p-jackson** [comment]
We're probably not going to be able to update the type in the DefinitelyTyped project because Gutenberg documents that type as a string. https://github.com/WordPress/gutenberg/blob/d923f2cae92f809541b7e009d68fd49ea3ea40ea/packages/components/src/tab-panel/README.md?plain=1#L121
But who knows, maybe we could submit a PR to change that documentation.

When I saw that the `<TabPanel>` component didn't support arbitrary elements for the title it made me question whether we'd be able to use this component 🤔 

But in the meantime, this is how I've hacked my way around "incorrect" type definitions in the past. Ideally we'd fix the issue upstream and then be able to get rid of this hack.

```typescript
interface MyTab extends Omit< TabPanel.Tab, 'title' > {
	title: React.ReactChild;
}

interface MyTabPanelProps extends Omit< TabPanel.Props, 'children' | 'tabs' > {
	tabs: readonly MyTab[];
	children: ( tab: MyTab ) => JSX.Element;
}

function MyTabPanel( { children, tabs, ...otherProps }: MyTabPanelProps ) {
	return (
		<TabPanel tabs={ tabs as any } { ...otherProps }>
			{ children as any }
		</TabPanel>
	);
}
```

And then instead of using `<TabPanel>` we use `<MyTabPanel>` (or whatever you'd rather call the component 😉)
Usually the hack doesn't take this many lines of code, but since we needed to change one of the component's sub-types it took a bit more.

Also noticed I used `ReactChild` instead of `ReactElement` for the title. That's because `ReactChild` will accept a string too.

File: packages/components/src/badge/index.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+	margin-left: 10px;
+	border-radius: 4px;
+	line-height: 20px;
+`;
+
+interface BadgePrpos {
+	children: ReactNode;
+}
+
+function Badge( { children }: BadgePrpos ) {
```
Comment (wojtekn) [inline packages/components/src/badge/index.tsx:19]: We are writing a component that has the same name as the existing [Badge Calypso UI component ](https://wpcalypso.wordpress.com/devdocs/design/badge). Will it be safe?

**Comment by wojtekn** [comment]
@sejas, I found an issue (or it may be it is still in progress) related to mobile view for tabs. On the medium mobile, I can't see the "Coming soon" tab, and I can't scroll through tabs. On the large mobile view, I see "Coming soon" partially, but it is broken into two lines:

Screen Shot 2022-07-05 at 11 23 53

File: packages/components/src/badge/index.tsx (hunk 1 above)
Comment (sejas) [reply packages/components/src/badge/index.tsx:19]: @wojtekn Thanks for finding the Badge. Yesterday I couldn't find that component.
I'll rename it and probably move it to `client/components`.

**Comment by sejas** [comment]
I've made a rebase of trunk and incorporated more changes than I wanted.
Closing this PR to start clean.

**Comment by sejas** [comment]
@p-jackson  @wojtekn I've addressed your feedback in a new PR https://github.com/Automattic/wp-calypso/pull/65296

**Pull request closed without merging**